### PR TITLE
shuf: add to osx platform

### LINKED ISSUE #2076: page request: shuf
shuf comes with gnu coreutils or something like that.

It can be installed both on Mac and Linux.

### PR DESCRIPTION
**Pull request by itsjzt**
shuf: add to the OSX platform. closes #2076

### CODE REVIEW

File: pages/osx/shuf.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+> Generate random permutations.
+
+- Randomize the order of lines in a file and output the result:
+
+`shuf {{filename}}`
+
+- Only output the first n entries of the result:
+
+`shuf -n {{n}} {{filename}}`
```
Comment (agnivade) [inline pages/osx/shuf.md]: Let's have a concrete no. like 5.

File: pages/osx/shuf.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+- Write output to another file:
+
+`shuf -o {{another_filename}} {{filename}}`
```
Comment (agnivade) [inline pages/osx/shuf.md]: s/another_filename/output_file

File: pages/osx/shuf.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+- Generate random numbers in range:
+
+`shuf -i {{low}}-{{high}}`
```
Comment (agnivade) [inline pages/osx/shuf.md]: Let's change low, high to 2-10. Or whatever suitable no. you think of.

**Comment by itsjzt** [comment]
@agnivade  should I change the same to Linux command too.

File: pages/osx/shuf.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+> Generate random permutations.
+
+- Randomize the order of lines in a file and output the result:
+
+`shuf {{filename}}`
+
+- Only output the first 5 entries of the result:
+
+`shuf -n 5 {{filename}}`
```
Comment (pxgamer) [inline pages/osx/shuf.md]: `5` should be in tokens (i.e. `{{5}}`)

File: pages/osx/shuf.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+- Write output to another file:
+
+`shuf -o {{another_filename}} {{filename}}`
+
+- Generate random numbers in range 1-10:
+
+`shuf -i 1-10`
```
Comment (pxgamer) [inline pages/osx/shuf.md]: `1-10` should also be in tokens.

File: pages/osx/shuf.md (hunk 2 above)
Comment (itsjzt) [reply pages/osx/shuf.md]: `{{another_filename}}` denotes actual file.

File: pages/osx/shuf.md (hunk 5 above)
Comment (itsjzt) [reply pages/osx/shuf.md]: like {1-10}

File: pages/osx/shuf.md [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+`shuf {{filename}}`
+
+- Only output the first 5 entries of the result:
+
+`shuf -n {{5}} {{filename}}`
+
+- Write output to another file:
+
+`shuf -o {{filename}} {{output_filename}}`
```
Comment (pxgamer) [inline pages/osx/shuf.md]: I thought this was correct with `-o {{output_file}} {{filename}}`? :)

Comment (itsjzt) [reply pages/osx/shuf.md]: oh you're right

**Comment by itsjzt** [comment]
I think after we finalize this `shuf` page then I will also change the Linux one accordingly

**Pull request merged**